test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java

```diff
@@ -21,116 +21,91 @@
  * questions.
  */
 
+import java.awt.AWTException;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Robot;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.awt.event.WindowEvent;
+
 /*
-  @test 1.2 98/08/05
+  @test
   @key headful
   @bug 4515763
   @summary Tests that clicking mouse and pressing keys generates correct amount of click-counts
-  @author andrei.dmitriev: area=awt.mouse
   @run main ClickDuringKeypress
 */
 
-/**
- * ClickDuringKeypress.java
- *
- * summary:
- */
-
-import java.applet.Applet;
-import java.awt.*;
-import java.awt.event.*;
-
-public class ClickDuringKeypress implements MouseListener
- {
+public class ClickDuringKeypress implements MouseListener {
    //Declare things used in the test, like buttons and labels here
    final static int CLICKCOUNT = 10;
-   final static int DOUBLE_CLICK_AUTO_DELAY = 10;
-   volatile int lastClickCount = 0;
-   volatile boolean clicked = false;
-   volatile boolean ready = false;
-
-   Frame frame;
-   Robot robot;
-
-   public void init()
-    {
-      //Create instructions for the user here, as well as set up
-      // the environment -- set the layout manager, add buttons,
-      // etc.
-
+   final static int DOUBLE_CLICK_AUTO_DELAY = 20;
+   static volatile int lastClickCount = 0;
+   static volatile boolean clicked = false;
+   static volatile boolean ready = false;
+
+   static volatile Frame frame;
+   static volatile Robot robot;
+   static final ClickDuringKeypress clicker = new ClickDuringKeypress();
```

Comment:
On the other hand, we can get rid of the instantiation of this and MouseListener empty methods overrides class by simply adding this:

```java
frame.addMouseListener(new MouseAdapter() {
          @Override
          public void mouseClicked(MouseEvent e) {
              System.out.println(e.toString());
              clicked = true;
              lastClickCount = e.getClickCount();
          }
      });
```

Either way, I'm fine with it.

Reply:
I thought about doing something like this .. but decided to stop the creep of changes.
So I'll leave it.